crates/core/src/db/datastore/system_tables.rs

```diff
@@ -208,13 +212,16 @@ st_fields_enum!(
     "allocated", Allocated = 8,
 });
 // WARNING: For a stable schema, don't change the field names and discriminants.
-st_fields_enum!(enum StConstraintFields {
-    "constraint_id", ConstraintId = 0,
-    "constraint_name", ConstraintName = 1,
-    "constraints", Constraints = 2,
-    "table_id", TableId = 3,
-    "columns", Columns = 4,
-});
+st_fields_enum!(
+    enum StConstraintFields {
+        "constraint_id", ConstraintId = 0,
+        "name", ConstraintName = 1,
+        // TODO(jgilles): can a system table store an enum?
+        // What if we need to add new variants to the enum? Will that break
+        // the schema? Does this need to store a byte array instead?
+        "data", ConstraintData = 3,
+    }
+);
```

Comment:
I would expect this to mirror how constraints are represented in `ModuleDef`. If we're going to have an enum here, that's what we should do in `ModuleDef` as well.

Notably postgres appears to put them all in a single table, but I would like you to confirm that.

Reply:
Ah JK you seem to have done that below.